src/pages/sym.mdx

```diff
@@ -0,0 +1,133 @@
+---
+
+import Link from '@docusaurus/Link';
+import { latestStable } from '@site/src/components/versions.js';
+
+export const LatestStableLink = ({children, to}) => <Link to={`/docs/${latestStable}/${to}`}>{children}</Link>;
+
+# Game Boy symbol (`.sym`) file format specification
+
+## Purpose & Context
+
+Game Boy ROM files contain pure binary data, but no metadata.
+In particular, symbol information is lacking when debugging or reverse-engineering such ROMs.
+Symbol files, or "sym files" for short, aim to be companions to Game Boy ROM files that provide exactly that: symbol information.
+
+Sym files have been around for a long time (the [no$gmb](http://problemkaputt.de/gmb.htm) emulator supported them, so it is safe to assume since at least the early 2000's), but the format was always loosely defined, leading to subtle incompatibilities.
+This document aims to standardize the format and usage of sym files in order to avoid this problem.
+Modifications, etc. should be suggested by [opening an issue](https://github.com/gbdev/rgbds-www/issues) or [starting a discussion](https://github.com/gbdev/rgbds-www/discussions) on this website's source repository, whichever is more appropriate.
+
+Extensions to this format are known to exist ([example](https://github.com/mattcurrie/mgbdis/#symbol-files)), but they tend to be specific to a certain use case, and are generally a backwards-compatible superset of what are described below.
+Keep in mind that some sym files are written manually, such as when disassembling ROMs, so it is recommended for consumers to be lenient in what they accept.
+
+Note also that [wla-dx's sym files](https://wla-dx.readthedocs.io/en/latest/symbols.html) are much more complex (and complete) than this format.
+RGBDS aims to keep sym files simple, and instead chooses to output other data in different files, such as map files[^1].
+
+[^1]: Map files remain thus far unspecified. Very few tools rely on information contained in them, so no specification has been written in order to keep that format extensible.
+
+## File format specification
+
+:::info
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+Numbers are written in decimal by default, or hexadecimal if prefixed with a `$`, unless otherwise specified.
+
+:::
+
+### Tokenization
+
+Line endings can be either LF-only or CRLF, though the former is preferred; they need not be consistent throughout the document.
+The character encoding must be well-formed UTF-8 (without BOM), however it is strongly encouraged to stick to ASCII characters for portability.
+The NUL character (`\0`, U+0000) shall not appear anywhere in the file.
+An implementation encountering one must output a diagnostic if it can, and may then ignore it, the line, the whole file, or reject the file outright.
```

Comment:
I don't think this needs to be specifically addressed. It's covered by these two lines:

- "If any part of a line cannot be processed by the implementation, it (and possibly the whole line) shall be ignored, and a warning should be produced."
- "The symbol name can contain any characters except ASCII spaces, tabs, non-printable ASCII characters, or characters excluded by earlier processing (~~semicolons, NULs, etc.~~ *e.g. semicolons*)."

The details of line processing, such as the definition of how the "location" is formatted, implicitly disallow NUL bytes; and  they're disallowed for symbol names because NUL is one of the non-printable ASCII characters.

Seeing the spec go into such low-level detail makes it sound like a more complicated format than it is. You could parse a standard-compliant file with a regex or two, or handle noncompliant ones with a `try { regex parse line } catch { ignore and warn }`.